### PR TITLE
[Android] add setSetVisibleCoordinateBounds api as already in iOS

### PR DESCRIPTION
**Pull request by uievolutionkk**
Added setSetVisibleCoordinateBounds api in Android.

This should partially solve the #2075.
https://github.com/mapbox/mapbox-gl-native/issues/2075

### CODE REVIEW

**Comment by tobrun** [comment]
Ran the code and looks good.
@ljbade can you validate the jni part?

I'm suggesting moving this to a separate feature fragment,
I will pick that up when merged.

cc @incanus @bleege

File: android/cpp/jni.cpp [hunk 1]
```diff
@@ -2059,4 +2198,4 @@ extern "C" JNIEXPORT void JNICALL JNI_OnUnload(JavaVM *vm, void *reserved) {
 
     theJVM = nullptr;
 }
-}
+}
```
Comment (ljbade) [inline android/cpp/jni.cpp]: Can you fix up the missing newline?

File: android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/geometry/CoordinateBounds.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    public LatLng getNe() {
+        return ne;
+    }
+
+    public void setNe(LatLng ne) {
+        this.ne = ne;
+    }
+}
```
Comment (ljbade) [inline android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/geometry/CoordinateBounds.java]: Same here.

File: android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/MapView.java [hunk 3]
```diff
@@ -1917,6 +1918,42 @@ private void deselectMarker() {
             }
         }
     }
+    
+    //
+    // Camera
+    //
+    public void setVisibleCoordinateBounds(@NonNull CoordinateBounds bounds) {
```
Comment (ljbade) [inline android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/MapView.java]: Can you JavaDoc for these new methods?

File: android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/NativeMapView.java [hunk 4]
```diff
@@ -578,6 +585,8 @@ private native void nativeSetBearing(long nativeMapViewPtr, double degrees,
     private native void nativeSetSprite(long nativeMapViewPtr, String symbol,
                                         int width, int height, float scale, byte[] pixels);
 
+    private native void nativeSetVisibleCoordinateBounds(long mNativeMapViewPtr, LatLng[] coordinates, RectF padding, double direction, long duration);
```
Comment (ljbade) [inline android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/views/NativeMapView.java]: You reformatted the other lines to align with the bracket but not this long line.

Either remove the reformat above or make this new line consistent.

File: android/java/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/MainActivity.java [hunk 5]
```diff
@@ -47,6 +52,7 @@
 
     // Used for saving instance state
     private static final String STATE_IS_ANNOTATIONS_ON = "isAnnotationsOn";
+
```
Comment (ljbade) [inline android/java/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/MainActivity.java]: Why are there extra newlines?

File: android/java/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/MainActivity.java [hunk 6]
```diff
@@ -58,9 +64,13 @@
 
     // Used for the UI
     private DrawerLayout mDrawerLayout;
+
```
Comment (ljbade) [inline android/java/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/MainActivity.java]: Same down here.

**Comment by ljbade** [comment]
@uievolutionkk Looking good. I added a few in-line comments.

Can you also add the other missing DPI bitmaps for the new crop icon you added?

File: android/java/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/MainActivity.java (hunk 5 above)
Comment (uievolutionkk) [inline android/java/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/MainActivity.java]: Seems they are due to the code formatting I used on Android Studio. Will fix it.

**Comment by Manabu-GT** [comment]
@ljbade Thank you for your feedback. I updated the PR based on your comments. Please take a look.

**Comment by ljbade** [comment]
@Manabu-GT thanks it looks great. We will merge this later this week once 2.1.0 is out the door.

**Comment by ljbade** [comment]
@Manabu-GT @uievolutionkk Now that I merged v2.1.0 and a few backlogged PRs, can you `git rebase master` this against current `master`.

Then we will be clear to merge.

**Comment by Manabu-GT** [comment]
@ljbade Done!

**Comment by tobrun** [comment]
@Manabu-GT 
merged! great work

**Pull request merged**